### PR TITLE
Added test cases for PR #11566

### LINKED ISSUE #13881: test case for syntax error on lambdify
See this issue https://github.com/sympy/sympy/issues/11044 . That issue was fixed in PR https://github.com/sympy/sympy/pull/11566. Though no test case was added for the `lambdify`.

#### What needs to be done?

Add test case that this works just fine:
```python
>>> from sympy import MatrixSymbol, lambdify
>>> x = MatrixSymbol('x', 3, 1)
>>> f = lambdify(x, x.T*x)
```

Test case needs to be added in appropriate file

### PR DESCRIPTION
**Pull request by lilithxxx**
Added test cases for https://github.com/sympy/sympy/pull/11566

Fixes https://github.com/sympy/sympy/issues/13881

### CODE REVIEW

File: sympy/utilities/tests/test_lambdify.py [hunk 1]
```diff
@@ -783,6 +783,13 @@ def test_issue_12092():
     f = implemented_function('f', lambda x: x**2)
     assert f(f(2)).evalf() == Float(16)
 
+def test_issue_13881(): # tests for #11566
+    from sympy import MatrixSymbol
+    x = MatrixSymbol('x', 3, 1)
+    f = lambdify(x, x.T*x)
+    assert f(Matrix([1, 2, 3])) == 14
```
Comment (Abdullahjavednesar) [inline sympy/utilities/tests/test_lambdify.py]: The test case here seems to fail.

Comment (lilithxxx) [reply sympy/utilities/tests/test_lambdify.py]: Yes I know but in my local terminal this test runs.

Comment (Abdullahjavednesar) [reply sympy/utilities/tests/test_lambdify.py]: I'm not sure why it fails here, I think you should try asking on gitter.

Comment (Abdullahjavednesar) [reply sympy/utilities/tests/test_lambdify.py]: Same was the issue with #13934.

Comment (czgdp1807) [reply sympy/utilities/tests/test_lambdify.py]: It's passing on my local terminal too.

**Pull request closed without merging**